src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/InstanceDataCollection.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Globalization;
+
+namespace System.Diagnostics
+{
+    /// <summary>
+    ///     A collection containing all the instance data for a counter.  This collection is contained in the 
+    ///     <see cref='System.Diagnostics.InstanceDataCollectionCollection'/> when using the 
+    ///     <see cref='System.Diagnostics.PerformanceCounterCategory.ReadCategory'/> method.  
+    /// </summary>    
+    public class InstanceDataCollection : DictionaryBase
+    {
+        private readonly string _counterName;
+
+        [Obsolete("This constructor has been deprecated.  Please use System.Diagnostics.InstanceDataCollectionCollection.get_Item to get an instance of this collection instead.  http://go.microsoft.com/fwlink/?linkid=14202")]
+        public InstanceDataCollection(string counterName)
+        {
+            if (counterName == null)
+                throw new ArgumentNullException(nameof(counterName));
+            _counterName = counterName;
+        }
+
+        public string CounterName
+        {
+            get
+            {
+                return _counterName;
+            }
+        }
+
+        public ICollection Keys
+        {
+            get { return Dictionary.Keys; }
+        }
+
+        public ICollection Values
+        {
+            get
+            {
+                return Dictionary.Values;
+            }
+        }
+
+        public InstanceData this[string instanceName]
+        {
+            get
+            {
+                if (instanceName == null)
+                    throw new ArgumentNullException(nameof(instanceName));
+
+                if (instanceName.Length == 0)
+                    instanceName = PerformanceCounterLib.SingleInstanceName;
+
+                object objectName = instanceName.ToLower(CultureInfo.InvariantCulture);
+                return (InstanceData)Dictionary[objectName];
+            }
+        }
+
+        internal void Add(string instanceName, InstanceData value)
+        {
+            object objectName = instanceName.ToLower(CultureInfo.InvariantCulture);
+            Dictionary.Add(objectName, value);
```

Comment:
I'm not sure why this compiles successfully? I believe you need a `@` before value if you want to use it as a parameter name.

Reply:
`value` is legal in context where it's unambiguous